File: src/SymmetricTree.java
public class SymmetricTree {

    public boolean isSymmetric(TreeNode root) {
        return root == null || isSymmetric(root.left, root.right);
    }

    public boolean isSymmetric(TreeNode p, TreeNode q) {
        if (p != null && q != null) {
            return p.val == q.val && isSymmetric(p.left, q.right) && isSymmetric(p.right, q.left);
        }
        return p == q;
    }
}

File: python/symmetric_tree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def is_symmetric(self, left: TreeNode, right: TreeNode) -> bool:
        if left and right:
            return left.val == right.val and self.is_symmetric(left.left, right.right) \
                   and self.is_symmetric(left.right, right.left)
        return left == right

    def isSymmetric(self, root: TreeNode) -> bool:
        return root is None or self.is_symmetric(root.left, root.right)
